Upstream form control styling for visionOS

megangardner opened the pull request:
#### 9c4012488b1d2c4d5711227cd10421adcde75682
<pre>
Upstream form control styling for visionOS
<a href="https://bugs.webkit.org/show_bug.cgi?id=257926">https://bugs.webkit.org/show_bug.cgi?id=257926</a>
rdar://110556097

Reviewed by Tim Nguyen.

* Source/WebCore/css/html.css:
(#if defined(ENABLE_DATE_AND_TIME_INPUT_TYPES) &amp;&amp; ENABLE_DATE_AND_TIME_INPUT_TYPES):
(#endif):
(input:is([type=&quot;button&quot;], [type=&quot;submit&quot;], [type=&quot;reset&quot;]), input[type=&quot;file&quot;]::file-selector-button, button):
(#endif // defined(ENABLE_INPUT_TYPE_COLOR) &amp;&amp; ENABLE_INPUT_TYPE_COLOR):
(textarea): Deleted.
* Source/WebCore/style/InspectorCSSOMWrappers.cpp:
(WebCore::Style::InspectorCSSOMWrappers::collectDocumentWrappers):
* Source/WebCore/style/UserAgentStyle.cpp:
(WebCore::Style::UserAgentStyle::ensureDefaultStyleSheetsForElement):
* Source/WebCore/style/UserAgentStyle.h:

Canonical link: <a href="https://commits.webkit.org/265054@main">https://commits.webkit.org/265054@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b5f96af7ec95e4cf3f4a56193527a34d512fd665

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/9634 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/9889 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/10134 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/11295 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/9415 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/9644 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/11869 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/9845 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/12336 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/9786 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/10638 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/8199 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/11454 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/7934 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/8756 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/16163 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/9028 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/8905 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/12240 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/9396 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/7631 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/8585 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/2315 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/12806 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/9170 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on Source/WebCore/css/html.css:
```diff
@@ -496,10 +496,15 @@ input:is([type="date"], [type="time"], [type="datetime-local"], [type="month"],
-    color: -apple-system-blue;
     padding: 0.2em 0.5em;
+#if defined(WTF_PLATFORM_VISION) && WTF_PLATFORM_VISION
+    background-color: rgba(0, 0, 0, 0.04);
```
This is a questionable choice (something like `color-mix(in srgb, CanvasText 4%, transparent)` would have been more accessible and given the same result), but this can be addressed separately.

aprotyas commented on Source/WebCore/css/html.css:
```diff
@@ -860,9 +861,13 @@ input:is([type="button"], [type="submit"], [type="reset"]), input[type="file"]::
+#if defined(WTF_PLATFORM_VISION) && WTF_PLATFORM_VISION
+    background-color: -apple-system-opaque-secondary-fill;
+#else
+    background-color: rgba(0, 0, 0, 0.05);
```
Wouldn't it be the other way around? I might be misreading though

nt1m replied:
yes

megangardner committed:
Upstream form control styling for visionOS
https://bugs.webkit.org/show_bug.cgi?id=257926
rdar://110556097

Reviewed by Tim Nguyen.

* Source/WebCore/css/html.css:
(#if defined(ENABLE_DATE_AND_TIME_INPUT_TYPES) && ENABLE_DATE_AND_TIME_INPUT_TYPES):
(#endif):
(input:is([type="button"], [type="submit"], [type="reset"]), input[type="file"]::file-selector-button, button):
(#endif // defined(ENABLE_INPUT_TYPE_COLOR) && ENABLE_INPUT_TYPE_COLOR):
(textarea): Deleted.
* Source/WebCore/style/InspectorCSSOMWrappers.cpp:
(WebCore::Style::InspectorCSSOMWrappers::collectDocumentWrappers):
* Source/WebCore/style/UserAgentStyle.cpp:
(WebCore::Style::UserAgentStyle::ensureDefaultStyleSheetsForElement):
* Source/WebCore/style/UserAgentStyle.h:

Canonical link: https://commits.webkit.org/265054@main
